### PR TITLE
core: deleted timestamps for chat item

### PR DESCRIPTION
**Pull request by epoberezkin**

### CODE REVIEW

File: src/Simplex/Chat/Messages.hs [hunk 1]
```diff
@@ -809,10 +815,7 @@ instance FromJSON MsgDecryptError where
 
 ciReactionAllowed :: ChatItem c d -> Bool
 ciReactionAllowed ChatItem {meta = CIMeta {itemDeleted = Just _}} = False
-ciReactionAllowed ChatItem {content} = case content of
-  CISndMsgContent _ -> True
-  CIRcvMsgContent _ -> True
-  _ -> False
+ciReactionAllowed ChatItem {content} = isJust $ ciMsgContent content
```
Comment (epoberezkin) [inline src/Simplex/Chat/Messages.hs:818]: refactoring

File: src/Simplex/Chat/View.hs [hunk 2]
```diff
@@ -493,38 +493,37 @@ viewItemNotChanged (AChatItem _ msgDir _ _) = case msgDir of
...
     prohibited = [styled (colored Red) ("[unexpected message deletion, please report to developers]" :: String)]
 
 viewItemReaction :: forall c d. Bool -> ChatInfo c -> CIReaction c d -> Bool -> CurrentTime -> TimeZone -> [StyledString]
 viewItemReaction showReactions chat CIReaction {chatDir, chatItem = CChatItem md ChatItem {chatDir = itemDir, content}, sentAt, reaction} added ts tz =
   case (chat, chatDir) of
-    (DirectChat c, CIDirectRcv) -> case content of
-      CIRcvMsgContent mc -> view from $ reactionMsg mc
-      CISndMsgContent mc -> view from $ reactionMsg mc
+    (DirectChat c, CIDirectRcv) -> case ciMsgContent content of
+      Just mc -> view from $ reactionMsg mc
```
Comment (epoberezkin) [inline src/Simplex/Chat/View.hs:520]: refactoring

File: src/Simplex/Chat/View.hs [hunk 3]
```diff
@@ -493,38 +493,37 @@ viewItemNotChanged (AChatItem _ msgDir _ _) = case msgDir of
...
       _ -> []
       where
         from = ttyFromContact c
         reactionMsg mc = quoteText mc $ if toMsgDirection md == MDSnd then ">>" else ">"
-    (GroupChat g, CIGroupRcv m) -> case content of
-      CIRcvMsgContent mc -> view from $ reactionMsg mc
-      CISndMsgContent mc -> view from $ reactionMsg mc
+    (GroupChat g, CIGroupRcv m) -> case ciMsgContent content of
+      Just mc -> view from $ reactionMsg mc
```
Comment (epoberezkin) [inline src/Simplex/Chat/View.hs:526]: refactoring

File: src/Simplex/Chat.hs [hunk 4]
```diff
@@ -2396,13 +2398,14 @@ deleteTimedItem user (ChatRef cType chatId, itemId) deleteAt = do
   ts <- liftIO getCurrentTime
   liftIO $ threadDelay' $ diffToMicroseconds $ diffUTCTime deleteAt ts
   waitChatStarted
+  deletedTs <- liftIO getCurrentTime
```
Comment (spaced4ndy) [inline src/Simplex/Chat.hs:2401]: can be taken in CTGroup branch

**Pull request merged**